src/clr-angular/forms/textarea/textarea-container.spec.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { FormsModule, NgControl } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+
+import { ClrIconModule } from '../../icon/icon.module';
+import { ClrCommonFormsModule } from '../common/common.module';
+import { IfErrorService } from '../common/if-error/if-error.service';
+
+import { ClrTextarea } from './textarea';
+import { ClrTextareaContainer } from './textarea-container';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { Layouts, LayoutService } from '../common/providers/layout.service';
+
+@Component({
+  template: `
+    <clr-textarea-container>
+      <textarea name="test" clrTextarea required [(ngModel)]="model"></textarea>
+      <label>Hello World</label>
+      <clr-control-helper>Helper text</clr-control-helper>
+      <clr-control-error>Must be at least 5 characters</clr-control-error>
+    </clr-textarea-container>
+    `,
+})
+class SimpleTest {
+  model = '';
+}
+
+@Component({
+  template: `
+  <clr-textarea-container>
+    <textarea clrTextarea [(ngModel)]="model"></textarea>
+  </clr-textarea-container>`,
+})
+class NoLabelTest {}
+
+export default function(): void {
+  describe('ClrTextareaContainer', () => {
+    let fixture, textareaContainerDE, textareaContainer, textareaContainerEl, ifErrorService, layoutService;
+
+    describe('no label', () => {
+      beforeEach(() => {
+        TestBed.configureTestingModule({
+          imports: [ClrIconModule, ClrCommonFormsModule, FormsModule],
+          declarations: [ClrTextareaContainer, ClrTextarea, NoLabelTest],
+          providers: [NgControl, NgControlService, IfErrorService, LayoutService],
+        });
+        fixture = TestBed.createComponent(NoLabelTest);
+
+        textareaContainerDE = fixture.debugElement.query(By.directive(ClrTextareaContainer));
+        textareaContainerEl = textareaContainerDE.nativeElement;
+        layoutService = textareaContainerDE.injector.get(LayoutService);
+      });
+
+      it('adds an empty label when instantiated without vertical layout', () => {
+        layoutService.layout = Layouts.HORIZONTAL;
```

Comment:
Maybe we should also test that the label **isn't** here when using a vertical layout?

Reply:
Done